dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -172,6 +174,13 @@ protected class PathVar[A](cast: String => Try[A]) {
       cast(str).toOption
     else
       None
+
+  def map[B](f: A => B): PathVar[B] =
+    new PathVar[B](str => cast(str).map(f))
+
+  def emap[B](f: A => Option[B]): PathVar[B] =
+    new PathVar[B](str => cast(str).flatMap(a => f(a).toRight(new NoStackTrace {}).toTry))
```

Comment:
A little bit of bikeshedding, but this feels like too many allocations for the simple instantiation... Considering the `PathVar` nature (something that's going to be used for every request on a given path), could we tweak this?